src/openrct2/world/Map.h

```diff
@@ -198,10 +198,16 @@ void map_invalidate_selection_rect();
 bool MapCheckCapacityAndReorganise(const CoordsXY& loc, size_t numElements = 1);
 TileElement* tile_element_insert(const CoordsXYZ& loc, int32_t occupiedQuadrants, TileElementType type);
 
+template<typename T = TileElement> T* MapGetFirstTileElementWithBaseHeightBetween(const TileCoordsXYRangedZ& loc)
```

Comment:
This isn't how i would expect this function to work. If you are specifying a type then you would expect it to find the first element of that type in the range.